Warning when applying config values

## Bug Report

<!--
Thanks for filing an issue! Before hitting the button, please answer these questions.
Fill in as much of the template below as you can. 
-->

**Describe the bug**
A clear and concise description of what the bug is.
I see a warning when I install K8ssandra with config values.

**To Reproduce**
Steps to reproduce the behavior:
In Katacoda I install K8ssandra as follows:
```$ helm repo add k8ssandra https://helm.k8ssandra.io/
$ helm repo update
$ helm install k8ssandra-cluster-a k8ssandra/k8ssandra   -f grafana-config-values.yaml
coalesce.go:199: warning: destination for prometheusSpec is a table. Ignoring non-table value <nil>
NAME: k8ssandra-cluster-a
LAST DEPLOYED: Mon Feb  1 21:35:24 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
```

Here's my config file:
```--- 
kube-prometheus-stack:
  prometheus:
    prometheusSpec:
      externalUrl: http://localhost:9090/prometheus
      routePrefix: /prometheus
    ingress:
      enabled: true
      paths:
        - /prometheus
  grafana:
    adminPassword: secret
    ingress:
      enabled: true
      path: /grafana
    plugins:
      - grafana-polystat-panel
    grafana.ini:
      server:
        root_url: http://localhost:3000/grafana
        serve_from_sub_path: true
```




**Expected behavior**
I did not expect to see the warning: `coalesce.go:199: warning: destination for prometheusSpec is a table. Ignoring non-table value <nil>`

**Screenshots**
N/A

**Environment (please complete the following information):**

* Helm charts version info
<!-- list installed charts and their versions from all namespaces -->
<!-- Replace the command with its output -->
```
$ helm ls -A
NAME                    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
k8ssandra-cluster-a     default         1               2021-02-01 21:35:24.191215676 +0000 UTC deployed        k8ssandra-0.36.0        3.11.7     
traefik                 default         1               2021-02-01 21:35:12.578829831 +0000 UTC deployed        traefik-9.14.0          2.4.0   
```

* Helm charts user-supplied values
<!-- For each k8ssandra chart involved list user-supplied values -->
<!-- Replace the commands with its output -->
```
$ helm get values k8ssandra-cluster-a
USER-SUPPLIED VALUES:
kube-prometheus-stack:
  grafana:
    adminPassword: secret
    grafana.ini:
      server:
        root_url: http://localhost:3000/grafana
        serve_from_sub_path: true
    ingress:
      enabled: true
      path: /grafana
    plugins:
    - grafana-polystat-panel
  prometheus:
    ingress:
      enabled: true
      paths:
      - /prometheus
    prometheusSpec:
      externalUrl: http://localhost:9090/prometheus
      routePrefix: /prometheus
```

```
$ helm get values k8ssandra-cluster-a
USER-SUPPLIED VALUES:
kube-prometheus-stack:
  grafana:
    adminPassword: secret
    grafana.ini:
      server:
        root_url: http://localhost:3000/grafana
        serve_from_sub_path: true
    ingress:
      enabled: true
      path: /grafana
    plugins:
    - grafana-polystat-panel
  prometheus:
    ingress:
      enabled: true
      paths:
      - /prometheus
    prometheusSpec:
      externalUrl: http://localhost:9090/prometheus
      routePrefix: /prometheus
$ helm get values traefik
USER-SUPPLIED VALUES:
ports:
  traefik:
    expose: true
    nodePort: 32090
  web:
    nodePort: 32080
  websecure:
    nodePort: 32443
providers:
  kubernetesCRD:
    namespaces:
    - default
  kubernetesIngress:
    namespaces:
    - default
service:
  type: NodePort
```

* Kubernetes version information:
<!-- Replace the command with its output -->
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2020-01-14T00:09:19Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:
<!-- Insert how you created your cluster: kind, kops, bootkube, etc. -->
```kind create cluster --config /root/kind-config.yaml```

```
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
networking:
  apiServerPort: 45451
nodes:
- role: control-plane
  kubeadmConfigPatches:
  - |
    kind: InitConfiguration
    nodeRegistration:
      kubeletExtraArgs:
        node-labels: "ingress-ready=true"
  extraPortMappings:
  - containerPort: 80
    hostPort: 80
    protocol: TCP
- role: worker
- role: worker
- role: worker
```




**Additional context**
<!-- Add any other context about the problem here. -->

--- a/charts/k8ssandra/Chart.yaml
+++ b/charts/k8ssandra/Chart.yaml
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.40.0
+version: 0.41.0
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to

--- a/charts/k8ssandra/values.yaml
+++ b/charts/k8ssandra/values.yaml
@@ -286,9 +286,12 @@ kube-prometheus-stack:
     # Provisions an instance of Prometheus as part of this release
     enabled: true
     prometheusSpec:
-      # Uncomment to listen for Prometheus traffic under a route prefix
-      # externalUrl: http://localhost:9090/prometheus
-      # routePrefix: /prometheus
+      # routePrefix is used to register routes, overriding externalUrl route. It is useful
+      # for proxies that rewrite URLs.
+      routePrefix: /
+
+      # externalUrl is an external URL at which Prometheus will be reachable.
+      externalUrl: ""
 
     # Enable templating of ingress resources for external prometheus traffic
     ingress:

